ssl/t1_lib.c

```diff
@@ -2333,6 +2333,14 @@ int tls_choose_sigalg(SSL *s, int *al)
 #else
                 continue;
 #endif
+            } else if (lu->sig == EVP_PKEY_RSA_PSS) {
+                /* validate that key is large enough for the signature algorithm */
+                const RSA *rsa = EVP_PKEY_get0_RSA(s->cert->pkeys[SSL_PKEY_RSA_PSS_SIGN].privatekey);
+                const EVP_MD *hash = EVP_get_digestbynid(lu->hash);
+                if ((NULL == rsa)
```

Comment:
Our coding style says to leave a blank line after declarations.  And we'd write the if test like this:
```
if (rsa == NULL
     || hash == NULL
     || RSA_size(rsa) < (2 * EVP_MD_size(hash) + 2))
```

Consider a new define RSA_PSS_KEY_BIG_ENOUGH or something, that captures that last less-than test.

Reply:
I'd be happy to add a define or function for the check.  Where is the best location for such a function?